projects/extension/sql/idempotent/001-openai.sql

```diff
@@ -19,196 +19,391 @@ set search_path to pg_catalog, pg_temp
 -- openai_detokenize
 -- decode tokens for a given model back into text
 -- https://github.com/openai/tiktoken/blob/main/README.md
-create or replace function ai.openai_detokenize(model text, tokens int[]) returns text
+create or replace function ai.openai_detokenize(_model text, _tokens int[]) returns text
 as $python$
     #ADD-PYTHON-LIB-DIR
     import tiktoken
-    encoding = tiktoken.encoding_for_model(model)
-    content = encoding.decode(tokens)
+    encoding = tiktoken.encoding_for_model(_model)
+    content = encoding.decode(_tokens)
     return content
 $python$
 language plpython3u strict immutable parallel safe security invoker
 set search_path to pg_catalog, pg_temp
 ;
 
+-------------------------------------------------------------------------------
+-- openai_client_create
+-- create the client and store it in the global dictionary for the session
+CREATE OR REPLACE FUNCTION ai.openai_client_create(
```

Comment:
What's the reason for exposing this functionality to the user? Seems like this should be an internal implementation detail.

Reply:
There are options when calling the client_create function that are unavailable as options to any of the specific endpoint calls. If you need to use them for your inference environment to work properly, there needs to be a way of configuring them.